lib/spandex_tesla_distributed.ex

```diff
@@ -48,8 +48,14 @@ defmodule SpandexTeslaDistributed do
         try do
           case Tesla.run(Tesla.put_headers(env, headers), next) do
             {:ok, env} ->
-              tracer.update_span(http: [status_code: env.status])
-              {:ok, env}
+              case env.status do
+                status when status not in 200..299 ->
+                  tracer.update_span(http: [status_code: env.status])
+                  tracer.span_error(%Error{message: "Request failed with status response #{env.status}"})
```

Comment:
```suggestion
                  tracer.span_error(%Error{message: "Request failed with status response #{env.status}"}, nil, [])
```

The other span_error's have the additional 2 params, not sure if they're required or not.

Reply:
Meant to add that back in 😅 Thanks for catching that!